clang/lib/Driver/Driver.cpp

```diff
@@ -5995,9 +6058,15 @@ InputInfo Driver::BuildJobsForActionNoCache(
     // FIXME: Clean this up.
     bool SubJobAtTopLevel =
         AtTopLevel && (isa<DsymutilJobAction>(A) || isa<VerifyJobAction>(A));
-    InputInfos.push_back(BuildJobsForAction(
-        C, Input, TC, BoundArch, SubJobAtTopLevel, MultipleArchs, LinkingOutput,
-        CachedResults, A->getOffloadingDeviceKind()));
+    if (auto *DA = dyn_cast<OffloadDepsJobAction>(A))
+      InputInfos.push_back(
+          BuildJobsForAction(C, Input, DA->getHostTC(), DA->getOffloadingArch(),
+                             SubJobAtTopLevel, MultipleArchs, LinkingOutput,
+                             CachedResults, DA->getOffloadingDeviceKind()));
+    else
+      InputInfos.push_back(BuildJobsForAction(
+          C, Input, TC, BoundArch, SubJobAtTopLevel, MultipleArchs,
+          LinkingOutput, CachedResults, A->getOffloadingDeviceKind()));
```

Comment:
NIt: could be made more compact to avoid repetition:

```suggestion
    auto *DA = dyn_cast<OffloadDepsJobAction>(A);
    InputInfos.push_back(
        BuildJobsForAction(C, Input, DA->getHostTC(), DA ? DA->getOffloadingArch() : BoundArch,
                           SubJobAtTopLevel, MultipleArchs, LinkingOutput,
                           CachedResults, DA->getOffloadingDeviceKind()));
```

Reply:
Done.